### PR TITLE
docs(bolt): Changes in examples of handshake.adoc

### PR DESCRIPTION
**Pull request by sagastume**
- Modification in the example to display the inclusion of version 4.1 instead of version 4.0.

- Adjustment in the example to accurately reflect the server's response with version 4.1.

### CODE REVIEW

File: modules/ROOT/pages/bolt/handshake.adoc [hunk 1]
```diff
@@ -105,13 +105,13 @@ The range cannot span multiple major versions.
 
 .Example with versions 4.3 plus two previous minor versions, 4.2 and 4.1
 ----
-00 02 03 04
+01 02 03 04
```
Comment (robsdedude) [inline modules/ROOT/pages/bolt/handshake.adoc:108]: The first byte is reserved. It's recommended to send 0, but the server will likely ignore it either way.
```suggestion
00 02 03 04
```

File: modules/ROOT/pages/bolt/handshake.adoc [hunk 2]
```diff
@@ -105,13 +105,13 @@ The range cannot span multiple major versions.
...
 ----
 
 .Example where the client is aware of five Bolt versions; 3, 4.0, 4.1, 4.2 and 4.3, and the server responds with 4.1
 ----
 C: 60 60 B0 17
-C: 00 03 03 04 00 00 01 04 00 00 00 04 00 00 00 03
+C: 00 02 03 04 00 00 01 04 00 00 00 04 00 00 00 03
```
Comment (robsdedude) [inline modules/ROOT/pages/bolt/handshake.adoc:114]: This change would mean the client requests `[4.3-4.1, 4.1, 4.0, 3.0]`, which is equivalent to the current `[4.3-4.0, 4.1, 4.0, 3.0]`. In fact 
```
C: 00 01 03 04 00 00 01 04 00 00 00 04 00 00 00 03
```
would be another equivalent client request.

I don't see a reason to change it, though. Finally, if we change it, we must not forget to update the note below explaining the versions requested.

Comment (sagastume) [reply modules/ROOT/pages/bolt/handshake.adoc:114]: Sorry for my clumsiness, but I can't understand why this "00 03 03 04" is equivalent to this: 4.3-4.1. Thank you very much.

File: modules/ROOT/pages/bolt/handshake.adoc (hunk 1 above)
Comment (sagastume) [reply modules/ROOT/pages/bolt/handshake.adoc:108]: Thank you for pointing that out. Although I don't recall reading that specification in the documentation, I think it's not relevant. However, I believe including the clarification about the first reserved byte could be useful for other developers. What do you think?"

Comment (robsdedude) [reply modules/ROOT/pages/bolt/handshake.adoc:108]: The fact that the first byte is reserved is documented right above this example

https://github.com/neo4j/docs-bolt/blob/d4576b0032ddb6742a3f7708cc20092ede707e26/modules/ROOT/pages/bolt/handshake.adoc?plain=1#L98

The value `00` is not documented other than in the example. The reason being that setting it to anything but 0, will likely cause no harm for servers following the version handshake scheme as described here. However, if we decide to make user of that byte, all code setting the byte to something else will have to be double-checked to not conflict with the future semantics.

It's not a rule set in stone to set it to 0, but we'll probably assume that it's 0 for legacy clients when/if we design some semantics. That's just because all official drivers set it to 0 and all the examples in the docs show it being 0 as well.

File: modules/ROOT/pages/bolt/handshake.adoc (hunk 2 above)
Comment (robsdedude) [reply modules/ROOT/pages/bolt/handshake.adoc:114]: No worries. I might have expressed myself unclear. Let me try again.

As per the docs, a version request of the form
```
00 02 03 04
```
means  Version `4.3` + the 2 consecutive versions before that, i.e., `4.1, 4.2, 4.3` = `4.1-4.3`.
That's obviously not equivalent to
```
00 03 03 04
```
which means `4.3-4.0`. However, with the following versions requested (the whole things reads):
```
C: 00 01 03 04   00 00 01 04   00 00 00 04   00 00 00 03
```
which translates to (parentheses to mark requested ranges): "server, please give me (4.3, 4.2), 4.1, 4.0 or 3.0 (is descending preference)".

Because of the overlap of the range with the explicitly requested versions, this is equivalent to
```
C: 00 03 03 04   00 00 01 04   00 00 00 04   00 00 00 03
```
which translates to: "server, please give me (4.3, 4.2, 4.1, 4.0), 4.1, 4.0 or 3.0 (is descending preference)".

File: modules/ROOT/pages/bolt/handshake.adoc (hunk 1 above)
Comment (sagastume) [reply modules/ROOT/pages/bolt/handshake.adoc:108]: Thank you very much for the response, it is clear now

File: modules/ROOT/pages/bolt/handshake.adoc (hunk 2 above)
Comment (sagastume) [reply modules/ROOT/pages/bolt/handshake.adoc:114]: Now I understand, I don't know why I always thought it was one version per byte. Now I have it clear, thank you very much.

**Comment by sagastume** [comment]
If you want, I can close the PR :) 
Thank you very much for the responses.
